src/payment/webhooks.py

```diff
@@ -0,0 +1,102 @@
+import logging
+from datetime import datetime
+
+import stripe
+from django.conf import settings
+from django.http import HttpResponse
+from django.utils import timezone
+from django.views.decorators.csrf import csrf_exempt
+
+from helpers.model_utils import get_object_or_400
+from payment.models import CustomerDetail, Subscription, User
+
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+stripe.api_key = settings.STRIPE_API_KEY
+STRIPE_WEBHOOK_SECRET = settings.STRIPE_WEBHOOK_SECRET
+
+
+@csrf_exempt
+def stripe_webhook(request):
+    payload = request.body
+    sig_header = request.META.get("HTTP_STRIPE_SIGNATURE")
+    event = None
+
+    try:
+        event = stripe.Webhook.construct_event(
+            payload, sig_header, STRIPE_WEBHOOK_SECRET
+        )
+    except ValueError:
+        return HttpResponse(status=400)
+
+    if event["type"] == "checkout.session.completed":
+        session = event["data"]["object"]
+        user = get_object_or_400(User, id=session["metadata"]["user_id"])
+        CustomerDetail.objects.update_or_create(
+            user=user,
+            defaults={
+                "stripe_customer_id": session["customer"],
+                "city": session["customer_details"]["address"]["city"],
+                "country": session["customer_details"]["address"]["country"],
+                "address": session["customer_details"]["address"]["line1"],
+                "postal_code": session["customer_details"]["address"]["postal_code"],
+                "email": session["customer_details"]["email"],
+            },
+        )
+        Subscription.objects.update_or_create(
+            user=user,
```

Comment:
Create a dedicated function for each case so that it's easier to read and test

Reply:
I understand your suggestion to create a dedicated function for each case, and you had mentioned this before. However, in this specific situation, I prefer keeping the code inline because:

Improved Readability in Context: The current approach makes it easy to understand each case without needing to navigate to multiple files or functions. For someone like me, still exploring Stripe, this minimizes context-switching and helps maintain focus.